### PR TITLE
[BACKLOG-5343]Log deprecation warning when using old ThinDriver

### PR DESCRIPTION
**Pull request by hudak**
@bmorrise @mkambol

### CODE REVIEW

File: core/src/org/pentaho/di/core/jdbc/ThinDriver.java [hunk 1]
```diff
@@ -67,6 +68,11 @@ public Connection connect( String url, Properties properties ) throws SQLExcepti
     String username = properties.getProperty( "user" );
     String password = properties.getProperty( "password" );
 
+    Logger.getLogger( getClass().getName() ).log( Level.WARNING,
+      getClass().getName() + " has been deprecated and is scheduled for removal in 6.1. "
+        + "Please use org.pentaho.di.trans.dataservice.jdbc.ThinDriver instead."
```
Comment (hudak) [inline core/src/org/pentaho/di/core/jdbc/ThinDriver.java]: @dkincade Message okay?

Comment (dkincade) [inline core/src/org/pentaho/di/core/jdbc/ThinDriver.java]: Yes ... but please add a @deprecated comments in this class using proper notation (http://wiki.pentaho.com/display/PEOpen/Deprecation+Guidelines)

File: core/src/org/pentaho/di/core/jdbc/ThinDriver.java [hunk 2]
```diff
@@ -29,12 +31,11 @@
...
 import java.sql.SQLFeatureNotSupportedException;
 import java.util.Properties;
+import java.util.logging.Level;
 import java.util.logging.Logger;
 
-import org.pentaho.di.core.KettleClientEnvironment;
-
 /**
- * This class is no longer used
+ * @deprecated As of release 6.0, use org.pentaho.di.trans.dataservice.jdbc.ThinDriver instead
```
Comment (hudak) [inline core/src/org/pentaho/di/core/jdbc/ThinDriver.java]: @dkincade updated javadoc

Comment (dkincade) [inline core/src/org/pentaho/di/core/jdbc/ThinDriver.java]: Perfect - thanks

**Pull request merged**